core/server/themes/index.js

```diff
@@ -45,5 +52,16 @@ module.exports = {
     loadOne: themeLoader.loadOneTheme,
     list: require('./list'),
     validate: validate,
-    toJSON: require('./to-json')
+    toJSON: require('./to-json'),
+    getActive: active.get,
+    activate: function activate(loadedTheme, checkedTheme) {
+        if (!_.has(checkedTheme, 'results.score.level') || checkedTheme.results.score.level !== 'passing') {
```

Comment:
A theme which has warnings gets a score level of "passing". Only a theme with errors will get a different level.